### PR TITLE
RFC: Add initial tool dependencies generator (WIP)

### PR DESCRIPTION
**Pull request by bgruening**
This will add a new script that can generates `tool_dependencies.xml` files automatically and downloads all packages as needed.

As I started to develop the `setup_*_environments` for the Tool Shed my aim was to completely isolate the installation from the rest of the system. We realised very quickly this was not the our main problem, the problem was to keep the packages persistent. Workarounds like https://github.com/bgruening/download_store are really hacky but became quite popular. I really appreciate this project and the idea and it will help us a lot.

Nevertheless, it does not solve the problem of R packages completely. There are many packages that are not stored in bioc and we need to keep them as well. Even worse, these packages may not be based on a VCS.

The attached script, still WIP, will download all packages that are needed by the produced package definition. We could offer a simple REST interface that will return a valid `tool_dependencies.xml` file. R packages, not part of bioc, will be downloaded and offered next to bioc packages with the aim that all packages, at the time of producing the tool dependency file, will be stored.

### CODE REVIEW

File: get_galaxy_tool_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    import rpy2.robjects as robjects
+    from rpy2.robjects.packages import importr
+    from rpy2.robjects.vectors import StrVector
+    import rpy2.robjects.packages as rpackages
+
+except ImportError:
+    raise ImportError(
+        "RPy2 must be installed to use this script.")
+
+ARCHIVE_URL_TEMPLATE = 'http://bioarchive.galaxyproject.org'
```
Comment (hexylena) [inline get_galaxy_tool_dependencies.py]: @erasche: change to https

**Comment by nturaga** [comment]
@bgruening Gonna leave this till after GCC/BOSC. But much needed, I remember Ross (fubar) had galaxy tool factory which does something similar, I can dig around once GCC is done.

**Comment by nturaga** [comment]
@bgruening We should talk about where the script fits in, and merge this. Thank you for the PR. This works very well.

**Comment by hexylena** [comment]
yeah, agreed. It's weird, it fits in between bioaRchive and the community package cache I feel like. It'd be a service that would be nice to offer in a place _next to_ (on the web) the CPC, so people building packages would have a one stop shop for caching URLs permanently, figuring out all of the requirements, etc

**Comment by bgruening** [comment]
@erasche I will be with @nitesh1989 next week and I think we will discuss all this in detail, to you mind joining for a video call?

**Comment by nturaga** [comment]
@erasche Yes, exactly! It does a lot of the work. We can probably modify a little to work out.

**Comment by hexylena** [comment]
@bgruening sounds great, video call it is! thanks for including me :) @nitesh1989 I'd love to see one of these for R/perl/etc.

**Comment by nturaga** [comment]
@erasche @bgruening Sounds like a plan!! I'll set it up and post a link when everyone is here.

**Comment by hexylena** [comment]
Is this now deprecated with conda stuff?

**Comment by bgruening** [comment]
I don't think so, tool_dependencies.xml files will stay for a while and this script should be extended to generate conda recipes.

**Comment by hexylena** [comment]
@bgruening OK, works for me.

**Pull request merged**